.github/actions/main-sample-app-build/action.yml

```diff
@@ -52,8 +90,17 @@ runs:
       run: |
         cp -r ${{ inputs.set-latest-main-build }} ${{ env.BUILD_FILE_NAME }}
 
+    # If the cache was not found, we need to copy the fallback build to the expected name.
+    # This is similar to the previous step, but we are doing it for the case where cache was not found.
+    - name: If cache was not found, copy fallback build to expected name
+      if: ${{ env.BUILD_CACHE_HIT == 'false' }}
+      working-directory: ${{ env.APN_SAMPLE_APP_BUILD_PATH }}
+      shell: bash
+      run: |
+        cp -r ${{ steps.fallback-build-main.outputs.app-xcarchive-name }} ${{ env.BUILD_FILE_NAME }}
+
     - name: Save latest main sample app build
-      if: ${{ inputs.set-latest-main-build != '' }}
+      if: ${{ inputs.set-latest-main-build != '' || env.BUILD_CACHE_HIT == 'false' }}
```

Comment:
One of the goals of this ticket was specifically to avoid looking for improvements at this stage. While there might be other areas we can improve, that's outside the current scope, and it's important to stay focused to avoid spending time on things outside of it.